### ISSUE TITLE
Foreign key columns within embedded entities are not prefixed

### ISSUE THREAD

**Issue by marshall007** [typeorm/typeorm#3132]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

```ts
@Entity()
export class ChangeSet {
  @PrimaryColumn('uuid')
  public id: string;

  @Column(() => ChangeSetAuthority)
  public initializer: ChangeSetAuthority;
  @Column(() => ChangeSetAuthority)
  public finalizer?: ChangeSetAuthority;
}

export class ChangeSetAuthority {
  @ManyToOne(() => Artifact)
  @JoinColumn({
    name: 'forArtifactId',
  })
  public artifact?: Artifact;
  @Column()
  public forArtifactId?: string;

  @ManyToOne(() => User)
  @JoinColumn({
    name: 'forUserId',
  })
  public user: User;
  @Column()
  public forUserId: string;
}
```

In the above scenario, duplicate definitions created for the relation ID columns with _different names_ and duplicate FK constraints are generated with the _same names_. See the following migration output as an example:

```
ALTER TABLE "change_sets" ADD "forArtifactId" uuid
ALTER TABLE "change_sets" ADD "forUserId" uuid
ALTER TABLE "change_sets" ADD "initializerForartifactid" character varying NOT NULL
ALTER TABLE "change_sets" ADD "initializerForuserid" character varying NOT NULL
ALTER TABLE "change_sets" ADD "finalizerForartifactid" character varying NOT NULL
ALTER TABLE "change_sets" ADD "finalizerForuserid" character varying NOT NULL
ALTER TABLE "change_sets" ADD CONSTRAINT "FK_f0e1be463c54638152058446def" FOREIGN KEY ("forArtifactId") REFERENCES "artifacts"("id")
ALTER TABLE "change_sets" ADD CONSTRAINT "FK_7c584ae18dac9a1d4fa42c2ab03" FOREIGN KEY ("forUserId") REFERENCES "users"("id")
ALTER TABLE "change_sets" ADD CONSTRAINT "FK_f0e1be463c54638152058446def" FOREIGN KEY ("forArtifactId") REFERENCES "artifacts"("id")
ALTER TABLE "change_sets" ADD CONSTRAINT "FK_7c584ae18dac9a1d4fa42c2ab03" FOREIGN KEY ("forUserId") REFERENCES "users"("id")
```

I was able to partially work around the first issue of duplicate column definitions by tweaking the naming strategy, but currently the FK constraint name generation does not take into account the possibility of embedded parent prefixes.

**Comment by pleerock** [comment]
yeah, so looks like the problem reproduces when you use embedded in conjunction with `@JoinColumn` decorator. Looks like not a hard-to-fix issue.

**Comment by gelito** [comment]
Is this issue still open?? I continue having this issue.

**Comment by Trezub** [comment]
Any workarounds to this?

**Comment by nebkat** [comment]
@Trezub it is solved in #6981 just waiting for merge now.
